### PR TITLE
Docs - Adds Local and Cloud dev workflows docs, create Workflow section of TOC

### PR DESCRIPTION
**Pull request by ernopp**
Adds two new workflow docs
* Local Dev workflow - google diagram [here](https://docs.google.com/drawings/d/1g7NNbHiTYKNAeu7X_BvwnpPgDBx5Op7MigwEpBP7T8s/edit) - paired on it with @m-samiec 
* Cloud Dev workflow - google diagram [here](https://docs.google.com/drawings/d/1JnpViwBGxIJa5S8bEN3nspaBKGNqmebZkLmgNdn2A_c/edit) - paired on it with @Vatyx 

The Google diagrams areembedded in improbadoc - this means they update live from the diagrams

2019-04-12 16_58_40-Window

2019-04-12 16_58_31-Window

Additionally renamed the Troubleshooting section of the docs to Workflow:

2019-04-12 16_58_48-Window

### CODE REVIEW

**Comment by m-samiec** [comment]
re: local workflow - I think the `incremental` step should be moved to after `stop deployment`, and both `Generate Schema` boxes lead into `Start Local Deployment`.

Also might be useful to have some visual flair between `Play Level` and `Stop Simulation` to indicate that's where you're actually running the game workers.

**Comment by ernopp** [comment]
Great points - incorporated in the diagram PTAL !

File: SpatialGDK/Documentation/content/cloud-dev-workflow.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+### Launch Cloud Deployment
+
+```
+spatial cloud launch --snapshot=snapshots/default.snapshot <assembly_name> launch_config.json <deployment_name>
+```
+
+Providing:
+
+* `<assembly_name>`, which identifies the worker assemblies to use (as chosen in the `spatial cloud upload` command).
```
Comment (ElleEss) [inline SpatialGDK/Documentation/content/cloud-dev-workflow.md]: Who chooses it? Our user? In which case:
```suggestion
* `<myassembly>` -  identifies the worker assemblies to use (as you specified earlier in the `spatial cloud upload` command). Replace `myassembly` with the name you choose to give your assembly.
```
Avoiding "which" means less text to read - nice in a list!  :-)
The explanation says "which assemblies to use" - does that mean you can specify more than one in the command? If not, we should say "assembly to use". If yes, we should say this explicitly.

File: SpatialGDK/Documentation/content/cloud-dev-workflow.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+```
+spatial cloud upload myassembly
+```
+
+Replacing `myassembly` with the name you choose to give your assembly.
+
+### Launch Cloud Deployment
+
+```
+spatial cloud launch --snapshot=snapshots/default.snapshot <assembly_name> launch_config.json <deployment_name>
```
Comment (ElleEss) [inline SpatialGDK/Documentation/content/cloud-dev-workflow.md]:
```suggestion
spatial cloud launch --snapshot=snapshots/default.snapshot <myassembly> launch_config.json <deployment_name>
```
Use the term consistent with above.

File: SpatialGDK/Documentation/content/cloud-dev-workflow.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+```
+
+Providing:
+
+* `<assembly_name>`, which identifies the worker assemblies to use (as chosen in the `spatial cloud upload` command).
+* `launch_config.json`, which declares the world and load balancing configuration.
```
Comment (ElleEss) [inline SpatialGDK/Documentation/content/cloud-dev-workflow.md]:
```suggestion
* `launch_config.json` - declares the world and load balancing configuration.
```
Avoiding "which" means less text to read - nice in a list! :-)

File: SpatialGDK/Documentation/content/cloud-dev-workflow.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+You may find the below snippets useful as reference:
+
+### Build server-worker assembly
+
+```
+Game\Plugins\UnrealGDK\SpatialGDK\Build\Scripts\BuildWorker.bat <YourProject>Server Linux Development <YourProject>.uproject
+```
+
+Replacing **`<YourProject>`** with the name of your Unreal project.
```
Comment (ElleEss) [inline SpatialGDK/Documentation/content/cloud-dev-workflow.md]:
```suggestion
Replacing `<YourProject>` with the name of your Unreal project.
```
No need for bold when we reference file names, code, command line arguments. Bold for UI references.

File: SpatialGDK/Documentation/content/cloud-dev-workflow.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+### Build client-worker assembly
+
+```
+Game\Plugins\UnrealGDK\SpatialGDK\Build\Scripts\BuildWorker.bat <YourProject> Win64 Development <YourProject>.uproject
+```
+
+Replacing **`<YourProject>`** with the name of your Unreal project.
```
Comment (ElleEss) [inline SpatialGDK/Documentation/content/cloud-dev-workflow.md]:
```suggestion
Replacing `<YourProject>` with the name of your Unreal project.
```

File: SpatialGDK/Documentation/content/cloud-dev-workflow.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+```
+spatial cloud launch --snapshot=snapshots/default.snapshot <assembly_name> launch_config.json <deployment_name>
+```
+
+Providing:
+
+* `<assembly_name>`, which identifies the worker assemblies to use (as chosen in the `spatial cloud upload` command).
+* `launch_config.json`, which declares the world and load balancing configuration.
+* `<deployment name>`, which labels the deployment in the [Console](https://console.improbable.io).
```
Comment (ElleEss) [inline SpatialGDK/Documentation/content/cloud-dev-workflow.md]:
```suggestion
* `<deployment name>`- labels the deployment for SpatialOS to reference in the [Console]({{urlroot}}/content/glossary#console).
```
Better in this circumstance to link to the glossary entry as it gives a little explanation on what the Console is before linking to it. A sort of documentation amuse-bouche.

File: SpatialGDK/Documentation/content/local-dev-workflow.md [hunk 7]
```diff
@@ -0,0 +1,9 @@
+# Local Development Workflow
+
+<!-- This is a live embed of a google drawing -->
+
+<img src="https://docs.google.com/drawings/d/e/2PACX-1vSCfvXabpgHPgNyb_9nr1rjcaHomevciDajOG1YNXol1a_qfa5Bll8swtncY_zcEiH1ZVMLH3v07PS8/pub?w=708&amp;h=1117">
+
+----
+
+Last updated: April 2019
```
Comment (ElleEss) [inline SpatialGDK/Documentation/content/local-dev-workflow.md]: _2019-04-15 Page added with editorial review_

File: SpatialGDK/Documentation/content/local-dev-workflow.md [hunk 8]
```diff
@@ -0,0 +1,9 @@
+# Local Development Workflow
+
```
Comment (ElleEss) [inline SpatialGDK/Documentation/content/local-dev-workflow.md]: It needs some sort of introduction and back-links to what local deployment means.

File: SpatialGDK/Documentation/toc.md [hunk 9]
```diff
@@ -28,7 +28,9 @@
     - [Directory structure]({{urlRoot}}/content/directory-structure)
     - [Gameplay Ability System]({{urlRoot}}/content/ability-system)
     - [Non-Unreal server-worker types]({{urlRoot}}/content/non-unreal-server-worker-types)
-- <h3>Troubleshooting</h3>
+- <h3>Workflow</h3>
+    - [Local Development Workflow]({{urlRoot}}/content/local-dev-workflow)
```
Comment (ElleEss) [inline SpatialGDK/Documentation/toc.md]:
```suggestion
    - [Local deployment workflow]({{urlRoot}}/content/local-dev-workflow)
```

File: SpatialGDK/Documentation/toc.md [hunk 10]
```diff
@@ -28,7 +28,9 @@
...
+    - [Cloud Development Workflow]({{urlRoot}}/content/cloud-dev-workflow)
```
Comment (ElleEss) [inline SpatialGDK/Documentation/toc.md]:
```suggestion
    - [Cloud deployment workflow]({{urlRoot}}/content/cloud-dev-workflow)
```

File: SpatialGDK/Documentation/toc.md [hunk 11]
```diff
@@ -28,7 +28,9 @@
     - [Directory structure]({{urlRoot}}/content/directory-structure)
     - [Gameplay Ability System]({{urlRoot}}/content/ability-system)
     - [Non-Unreal server-worker types]({{urlRoot}}/content/non-unreal-server-worker-types)
-- <h3>Troubleshooting</h3>
+- <h3>Workflow</h3>
```
Comment (ElleEss) [inline SpatialGDK/Documentation/toc.md]:
```suggestion
- <h3>Workflows</h3>
```

???

**Comment by ElleEss** [comment]
Have pontificated muchly on this baby: https://docs.google.com/drawings/d/1g7NNbHiTYKNAeu7X_BvwnpPgDBx5Op7MigwEpBP7T8s/edit

File: SpatialGDK/Documentation/toc.md (hunk 9 above)
Comment (ElleEss) [inline SpatialGDK/Documentation/toc.md]:
```suggestion
    - [Local development workflow]({{urlRoot}}/content/local-dev-workflow)
```

File: SpatialGDK/Documentation/toc.md (hunk 10 above)
Comment (ElleEss) [inline SpatialGDK/Documentation/toc.md]:
```suggestion
    - [Cloud development workflow]({{urlRoot}}/content/cloud-dev-workflow)
```

Comment (ElleEss) [reply SpatialGDK/Documentation/toc.md]: I was confused between "development" and "deployment" first time round.

(Defo needs lower case initial letters)

File: SpatialGDK/Documentation/toc.md (hunk 9 above)
Comment (ElleEss) [reply SpatialGDK/Documentation/toc.md]: I was confused between "development" and "deployment" first time round.
(Defo needs lower case initial letters)

**Comment by ElleEss** [comment]
Have also pontificated on: https://docs.google.com/drawings/d/1JnpViwBGxIJa5S8bEN3nspaBKGNqmebZkLmgNdn2A_c/edit

File: SpatialGDK/Documentation/content/cloud-dev-workflow.md [hunk 12]
```diff
@@ -0,0 +1,47 @@
+# Cloud Development Workflow
+
+<!-- This is a live embed of a google drawing -->
```
[the remark this answers is not in the document]
Comment (ernopp) [reply SpatialGDK/Documentation/content/cloud-dev-workflow.md]: yep in develop drive and with big warning

[the remark this answers is not in the document]
Comment (ernopp) [reply SpatialGDK/Documentation/content/cloud-dev-workflow.md]: https://drive.google.com/drive/folders/1FugEDuIHe6Dy_ZSi-FawEumHSgtcbDoZ

File: SpatialGDK/Documentation/content/cloud-dev-workflow.md (hunk 6 above)
Comment (ernopp) [reply SpatialGDK/Documentation/content/cloud-dev-workflow.md]: good point, incorporated

**Comment by ElleEss** [comment]
Diagram for Local Loads better. ("Full (MRI) Scan" though. 
https://docs.google.com/drawings/d/1g7NNbHiTYKNAeu7X_BvwnpPgDBx5Op7MigwEpBP7T8s/edit

**Comment by ElleEss** [comment]
Diagram for Cloud is loads better. A couple of comments.
https://docs.google.com/drawings/d/1JnpViwBGxIJa5S8bEN3nspaBKGNqmebZkLmgNdn2A_c/edit

**Pull request merged**